core/src/main/scala/cats/derived/show.scala

```diff
@@ -21,23 +21,16 @@ object MkShow extends MkShowDerivation {
   def apply[A](implicit show: MkShow[A]): MkShow[A] = show
 }
 
-trait MkShowDerivation extends MkShow0 {
-
-  implicit def fromShow[T](implicit s: Show[T]): MkShow[T] = new MkShow[T] {
-    override def show(t: T): String = s.show(t)
-  }
-}
-
-trait MkShow0 extends MkShow1 {
+trait MkShowDerivation extends MkShow1 {
   implicit val emptyProductDerivedShow: MkShow[HNil] =
     instance(_ => "")
 
   implicit def productDerivedShowFurther[K <: Symbol, V, T <: HList](
        implicit key: Witness.Aux[K],
-       showV: Lazy[MkShow[V]],
+       showV: Show[V] OrElse MkShow[V],
```

Comment:
I think we should do this for coproducts as well. Although less likely that someone would define a special case only for one subclass, it's still a possibility.

Reply:
thanks I missed that.